### PR TITLE
Make ato build again - Part 3 the Last Crusade

### PR DESCRIPTION
**Pull request by mawildoer**

### CODE REVIEW

File: src/atopile/_shim.py [hunk 1]
```diff
@@ -47,7 +41,7 @@ def _is_int(name: str) -> bool:
     return True
 
 
-class _has_kicad_footprint_name_defined(F.has_footprint_impl):
+class has_local_kicad_footprint_named_defined(F.has_footprint_impl):
```
Comment (mawildoer) [inline src/atopile/_shim.py:44]: Hopefully we can replace this from this shim momentarily w/ your new explicit part pick

File: src/atopile/_shim.py [hunk 2]
```diff
@@ -238,7 +249,7 @@ def value(self) -> L.Range:
 
     @value.setter
     def value(self, value: L.Range):
-        _alias_is(self.capacitance, value)
```
Comment (mawildoer) [inline src/atopile/_shim.py:241]: Better models existing ato behaviour

File: src/atopile/buildutil.py [hunk 3]
```diff
@@ -21,6 +25,7 @@
     export_pick_and_place,
     export_step,
 )
+from faebryk.exporters.pcb.kicad.pcb import _get_footprint
```
Comment (mawildoer) [inline src/atopile/buildutil.py:28]: Used as a temp measure to check footprint quality early and avoid late (eg. long) expensive failures

File: src/faebryk/exporters/pcb/kicad/pcb.py [hunk 4]
```diff
@@ -319,83 +333,89 @@ def get_property_uuid(
...
-
-            at = C_xyr(x=x, y=y, r=0)
-
-            # separate components vertically within group
-            y += VERTICAL_SPACING
-
-            if comp_name in comps_changed:
-                # TODO also need to do geo rotations and stuff
-                at = comps_changed[comp_name].at
+        for err_collector, comp_name in iter_through_errors(
```
Comment (mawildoer) [inline src/faebryk/exporters/pcb/kicad/pcb.py:336]: Dunno why the diff aligns so poorly. This should all just be indented one to add the error collection context. Double checking now

File: src/faebryk/libs/kicad/fileformats.py [hunk 5]
```diff
@@ -1200,7 +1200,7 @@ class C_kicad_footprint_file(SEXP_File):
     class C_footprint_in_file(C_footprint):
         descr: Optional[str] = None
         tags: Optional[list[str]] = None
-        version: int = field(**sexp_field(assert_value=20240108), default=20240108)
```
Comment (mawildoer) [inline src/faebryk/libs/kicad/fileformats.py:1203]: Shit, should re-add this. Will fix

File: src/faebryk/libs/sexp/dataclass_sexp.py [hunk 6]
```diff
@@ -99,24 +100,36 @@ def _convert(
         if (origin := get_origin(t)) is not None:
             args = get_args(t)
             if origin is list:
-                return [_convert(_val, args[0], substack) for _val in val]
+                return [
```
Comment (mawildoer) [inline src/faebryk/libs/sexp/dataclass_sexp.py:103]: This is as a backstop to TRY load stuff. I think we could take a more delicate approach in the future, but works well for now. See https://github.com/atopile/atopile/pull/711/files#diff-02f2b52a27305bdc76f47e7e4e5e3040a17a3b9b12bcb4cb3593b245e028e6a9

File: src/faebryk/exporters/pcb/kicad/pcb.py (hunk 4 above)
Comment (mawildoer) [reply src/faebryk/exporters/pcb/kicad/pcb.py:336]: Yep, confirmed there are no other changes

**Review by iopapamanoglou — APPROVED** [review]

**Pull request merged**